tutorials/mnist_tutorial_tf.py

```diff
@@ -24,11 +24,18 @@
 flags.DEFINE_float('learning_rate', 0.1, 'Learning rate for training')
 
 
-def main(argv=None):
+def mnist_tutorial(testing=False):
     """
     MNIST cleverhans tutorial
+    :param testing: only used internally for cleverhans testing purposes
     :return:
     """
+    # Used for cleverhans testing purposes only. Content returned must be:
+    # [0] accuracy of model on clean data
+    # [1] accuracy of model on adversarial examples
+    # [2] accuracy of adversarially trained model on clean data
+    # [3] accuracy of adversarially trained model on adversarial examples
+    test = np.zeros(4, dtype=np.float32)
```

Comment:
how about something like

    class AccuracyReport(object):
        """
        An object summarizing the accuracy results for experiments involving training on clean examples or adversarial examples, then evaluating on clean or adversarial examples.
        """
        def __init__(self, clean_train_clean_eval, clean_train_adv_eval, adv_train_clean_eval, adv_train_adv_eval):
           self.__dict__.update(locals())
           del self.self